### ISSUE TITLE
Improvments

### ISSUE THREAD

**Issue by alexdor** [alexdor/dtu-ai-mas-final-assignment#16]
## 1. Sanity checks
- [x] What happened to the 3 levels we lost between master and MA implementation
- [x] Test state expansion

## 2. Preproc
- [x] Turn "dead" boxes into walls, when there are no agents to move the box
- [ ] Figure out box to goal solution (?)

## 3. Multi agents
- [ ] Trim state expansion for multi agent
- [x] Fix multi-agent conflict resolution (add CBS)
- [ ] Fix CBS implementation

## 4. Heuristics
- [ ] When the wall cost is zero (between the box and the goal) don't expand directions that don't get the box closer to the goal (note: The agent shouldn't be between the goal and the box in order for this to apply)

## 5. Preproc multi only
- [x] Assign agents to boxes (don't search each time)
- [ ] Subgoal that clears the path for another agent

## 6. Overall perf
- [x] Drop a lot of pointers

## 7. Nice to have
- [ ] Use bi direction search instead of A*

**Comment by alexdor** [comment]
@Westh @s153986 do we have a lvl where the agent has to be in a goal state in order for the lvl to complete? Because we don't have that as a planned issue

**Comment by Westh** [comment]
> @Westh @s153986 do we have a lvl where the agent has to be in a goal state in order for the lvl to complete? Because we don't have that as a planned issue

yeah `SAHoldud.lvl` is one such example, haven't actually seen if there are more - but I am thinking that we could just alter those levels to so that they *actually* comply with the rules, and then we can always mention that in the report 😄
